core/src/client/mod.rs

```diff
@@ -183,6 +185,11 @@ pub trait Shard {
     fn shard_root(&self, shard_id: ShardId, state: StateOrBlock) -> Option<H256>;
 }
 
+pub trait ResealTimer: TimeoutHandler {
```

Comment:
It's because `NetworkExtension` is tightly coupled with `TimerHandler`.
[Client::register_extension](https://github.com/CodeChain-io/codechain/blob/master/network/src/client.rs#L127) uses the extension as `TimeoutHandler`.
But the `ResealTimer` can be decoupled with `TimeoutHandler`.

Reply:
The core client in this patch is also used as `TimeoutHandler`.

Reply:
I've deleted `TimeoutHandler` from `ResealTimer`